nemo/core/classes/common.py

```diff
@@ -240,11 +240,17 @@ def __attach_neural_type(self, obj, type_val, name=None):
 
 class Serialization(ABC):
     @classmethod
-    def from_config_dict(cls, config: DictConfig, *args: Any, **kwargs: Any):
+    def from_config_dict(cls, config: DictConfig):
         """Instantiates object using DictConfig-based configuration"""
+        # Resolve the config dict
+        if isinstance(config, DictConfig):
+            config = OmegaConf.to_container(config, resolve=True)
+            config = OmegaConf.create(config)
+            OmegaConf.set_struct(config, True)
+
         if ('cls' in config or 'target' in config) and 'params' in config:
             # regular hydra-based instantiation
-            instance = hydra.utils.instantiate(config=config, *args, **kwargs)
+            instance = hydra.utils.instantiate(config=config)
         else:
             # models are handled differently for now
             # TODO: allow passthrough for args, and kwargs too?
```

Comment:
maybe remove this TODO since people keep adding it :D

Reply:
Haha there is the offending line lol

Reply:
Why not add override support for models instead of ignoring it?

Reply:
The entire yaml would be wrapped in a target, params block and that breaks most assumptions about structure of yaml. So models themselves woudk need a major rewrite of core to support a one pattern, which would just shift all arguments one layer deeper into the hierarchy when overriding. I am not in support of this. Makes it much more verbose to edit arguments from command line with no actual benefit.

Reply:
Ok, so can we split from_config_dict into two classes? One that uses hydra and one that doesnt?

Reply:
from_config_dict does use hydra if it finds that the config has `cls` or `target`. You cannot actually use `cls` and `target` inside `model` namespace of yaml. Optimizers, schedulers, data loaders for all 3 sets, basically all of core would break if you tried it. 

Its not just an issue with from_config_dict. Our code parsing makes the assumption that inside `model`, yes there can be some standard uses of cls and params, but model itself cannot be wrapped inside cls and params. This was done by design, when we tried it with optimizers and schedulers, the command line override was far too deep. Pushing them one more level up due to `model`.cls and model.params would just be unusable.

Reply:
Ok what I am hearing if that you don't want to support `*args, **kwargs` inside of `from_config_dict`, nor do you want to split the function into two separate functions. That's fine, if `from_config_dict` doesn't serve my purposes, I will use `hydra.utils.instantiate` directly instead.
I will change the TTS collection in a separate PR.